### PR TITLE
Dockerfile updates and test

### PR DESCRIPTION
**Pull request by richard-rogers**
## Description

Update protobuf version used in the `Dockerfile.dev` and test the image.

- [ ] I have reviewed the [Guidelines for Contributing](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md).

### CODE REVIEW

File: .github/workflows/docker-test.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: Nightly Docker test
```
Comment (FelipeAdachi) [inline .github/workflows/docker-test.yml:1]: maybe in the future we could figure out api keys in github to run periodic integration tests!

Comment (FelipeAdachi) [reply .github/workflows/docker-test.yml:1]: resolving this as it's not an actual request, just a comment

**Review by jamie256 — DISMISSED** [review]
LGTM!

**Review by jamie256 — DISMISSED** [review]
LGTM!

**Review by jamie256 — APPROVED** [review]

**Pull request merged**